src/mcode-pilot/utils/recordToProps.js

```diff
@@ -174,7 +174,8 @@ function _mapProp(propDict) {
             const option = propDict[key][prop];
             // drops option boxes that don't have
             // a value from the patient record
-            if (option.value) {
+            if ((filters!==undefined && filters.includes(option.mcodeElement) && option.value !== null && option.value !== undefined) ||
```

Comment:
Can we make this a function in the file so that we don't have like so many boolean checks on one line? Or at least can we comment this here to explain what we're checking for here. There's a lot of logic nested in one space.